SPECIFICATION.md

```diff
@@ -0,0 +1,143 @@
+# The FontoBene Font Format, Version 1.0
+
+
+## Introduction
+
+FontoBene is a vector font format. It only uses strokes to form glyphs, in
+order for easy rendering in systems that cannot easily deal with more complex
+parts of formats like TTF or OTF, like Bézier curves.
+
+FontoBene was designed with the use case of text on printed circuit boards
+(PCBs) in mind. PCBs are usually manufactured using the Gerber format as an
+exchange format to describe the etching layers, which does not support
+quadratic / cubic curves.
+
+The main inspiration for this font format was the LFF font used by LibreCAD.
+
+
+## High Level Format Description
+
+Every FontoBene font file consists of UTF-8 encoded text. The lines are
+separated by newlines (U+000A LINE FEED). It starts with a header, followed by
+glyph definitions.
+
+
+### Header
+
+The header is parsed line-by-line, starting with a number sign (`#`) as the
+first character on every line. The number sign is followed by a space, a key,
+and a value. Key and value are separated by a colon (`:`). Everything following
+the colon is considered the value. Both keys and values are strings.
+
+Example:
+
+    # Format: FontoBene/1.0
+    # Creator: Max Müller
+    # Version: 1.0
+    # VersionDate: 2017-09-27
+    # License: GPLv3+
+
+All keys that may be used are specified in the specification. A user MAY use
+non-standard keys, but must prefix them with `X-`, for example `X-Foo: Bar`.
```

Comment:
Not sure if the `X-` is a good idea. I'd like to have a way to discern standardized from non-standard headers though. Otherwise it could happen that users start defining non-standard headers that become standardized later, resulting in invalid header entries, making the font files non-forwards-compatible.

Reply:
Why not write non-standardized data as comments, where arbitrary text can be written anyway?

Or if we use a section `[meta]` as described above, we could define a section `[meta_nonstandard]` or something like that, which could be used for non-standardized metadata.